lms/services/jstor.py

```diff
@@ -0,0 +1,28 @@
+from lms.views.helpers import via_url
+
+
+class JSTORService:
+    def __init__(self, settings):
+        self._settings = settings
+
+    @property
+    def enabled(self):
+        return self._settings.get("enabled", False)
+
+    @staticmethod
+    def via_url(request, document_url):
+        return via_url(
+            request,
+            document_url,
+            content_type="pdf",
+            options={"jstor.ip": request.registry.settings["jstor_ip"]},
+        )
+
+
+def factory(_context, request):
+    # application_instance.settings is an ApplicationSettings which overwrites `get`
+    # convert to a dict to be able to access ["jstor"] securely
```

Comment:
I couldn't parse the part of this comment after "overwrites `get`". Can you clarify why wrapping `ApplicationSettings.get` is needed here?

Reply:
As `ApplicationSettings.get` is implemented it only works to access two nested levels.

"get("jstor", "enabled", False)" - Returns False when either "jstor" or "jstor->enable" don't exist.

Here we want the full `jstor` dict. But ApplicationSettings doesn't allow for just one level, ie you can't do get("jstor", default=False).

Having "key" be optional here https://github.com/hypothesis/lms/blob/1b6049dc81d9287ab34fd78efe1eac56ed844f96/lms/models/application_settings.py#L35 might be the right solution but I rather tackle that on it's own.

Reply:
Ah - I see why I didn't understand the comment! I didn't realize these were two separate sentences (ending after `get` and beginning with `Convert`). Can you update this and then it is good to go.